packages/repository/src/repositories/inclusion.ts

```diff
@@ -0,0 +1,96 @@
+import {Entity} from '../model';
+import {Getter} from '@loopback/core';
+import {
+  Filter,
+  Where,
+  resolveHasManyMetadata,
+  resolveBelongsToMetadata,
+  RelationMetadata,
+  RelationType,
+  constrainWhere,
+} from '../';
+import {AnyObject} from '..';
+import {DefaultCrudRepository} from './legacy-juggler-bridge';
+import {inspect} from 'util';
+import {
+  HasManyDefinition,
+  BelongsToDefinition,
+  HasManyResolvedDefinition,
+  BelongsToResolvedDefinition,
+} from '../relations';
+
+type ResolvedRelationMetadata =
+  | HasManyResolvedDefinition
+  | BelongsToResolvedDefinition;
+
+// SE: the source entity
+// TE: the target entity
+// SID: the ID of source entity
+// TID: the ID of target entity
+
+export class InclusionHandler<SE extends Entity, SID> {
+  _handlers: {[relation: string]: Function} = {};
+  constructor(public sourceRepository: DefaultCrudRepository<SE, SID>) {}
+
+  registerHandler<TE extends Entity, TID>(
+    relationName: string,
+    targetRepoGetter: Getter<DefaultCrudRepository<TE, TID>>,
+  ) {
+    this._handlers[relationName] = fetchIncludedItems;
+    const self = this;
+
+    async function fetchIncludedItems(
```

Comment:
It can be simplified as:
```ts
this._handlers[relationName] = async (
      fks: SID[],
      filter?: Filter<TE>,
    ): Promise<TE[]> => {...}
```

Reply:
Maybe we should define a type such as:

```
export type IncludedItemsFetcher<SID, TE> = (
      fks: SID[],
      filter?: Filter<TE>,
    ) => Promise<TE[]>
```